scripts/fabric_links_monitoring.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python2.7
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+import time
+import json
+import syslog
+import subprocess
+
+from swsssdk import  SonicV2Connector, SonicDBConfig
+from sonic_py_common import multi_asic
+
+class FabricLinkErrMontoring():
+    '''Class to monitor the fabric link errors for a given namespace'''
+
+    def __init__(self, namespace, port_name_map, filter_for_up_ports=None):
+        '''Initialize the class with the namespace and port_name_map in counters db'''
+        self.namespace = namespace
+        self.port_name_map = port_name_map
+        self.filter_for_up_ports = filter_for_up_ports
+        self.port_map = self.get_port_map()
+        print('Fabric device id :', namespace)
+        subprocess.run(["sonic-clear", "fabriccountersport"])
+        self.prev_port_counters = self.port_counters
+
+    @property
+    def counter_db(self):
+        '''Return the counter db object'''
+        db = SonicV2Connector(namespace=self.namespace)
+        db.connect('COUNTERS_DB')
+        return db
+
+    def get_port_map(self):
+        '''
+        Return the port map from the counter db
+        for the port_names in the port_name_map
+        Additonally the port map can be filtered for UP ports
+        '''
+        ports = self.counter_db.get_all('COUNTERS_DB', self.port_name_map)
+        if self.filter_for_up_ports:
+            port_to_mon = self.filter_for_up_ports(self.namespace, ports)
+        return port_to_mon
```

Comment:
Don't you need to return ports if self.filter_for_up_ports is not defined.

Reply:
if ports are down then it port to monitor may be empty.